docs/rfcs/1008-nightly-build.md

```diff
@@ -0,0 +1,206 @@
+
+> We could also [list all releases] to find the one we look for.
+
+[list all releases]: https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#list-releases
+
+So we could update the workflow to directly create the release.
+
+> At this step, you may be asking yourself [Why do we need to release the nightly build ?](#why-do-we-need-to-release-the-nightly-build-or-why-we-cant-directly-use-the-artifact-link-of-a-workflow)
+
+### Snapcraft channel with tracks
+
+Our currently workflow with Snapcraft we only use the default channels `stable` & `edge`.
+
+I would like to introduce a [track] in the channel.
+
+A track allow to more precisely specify which version to install with `snap`, what I have currently in mind is the following track:
+
+- `latest`: We already have this one since it's the default one used by Snapcraft when we don't specify it
+- `v{MAJOR}`: A track per major version (in the case we need to maintained multiple version at the same time)
+- `insider`: A track that contain the nightly build
+
+  > I've also thought of the following track name instead of `insider` (VS-Code style):
+  > - `canary` (Chrome style)
+  > - `nightly` (Firefox style)
+
+[track]: https://snapcraft.io/docs/channels#heading--tracks
+
+That would translate to the following usage:
+
+```shell
+snap install parsec # This will use `latest/stable`
+snap install parsec --channel=latest # will use `latest/stable`
+snap install parsec --channel=latest/beta # will use `latest/beta`
+snap install parsec --channel=v3 # will use `v3/stable`
+snap install parsec --channel=insider # will use `insider/stable`
+```
+
+We could simply put the versions that contain a `dev` part into the `insider` track.
+
+### Schedule the workflow
+
+Require [Overwriting nightly release & automatic release](#overwriting-nightly-release--automatic-release)
+
+We need to have a workflow that schedule the nightly build at a specific interval (every day at 22H30 for example).
+GitHub already allow to [trigger a workflow on a schedule].
+
+[trigger a workflow on a schedule]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule
+
+> [!CAUTION]
+> Event triggered by a GitHub actions don't trigger a workflow. So everything should be done in a single workflow.
+
+### Slack notification (optional)
+
+We could send a Slack notification when a nightly build finished (successfully or not)
+With the link to download the generated build.
+
+Slack provide GitHub action for that: <https://github.com/slackapi/slack-github-action>
+
+### Improve the unique dev format
+
+Currently, `releaser.py` generate a dev version by appending the `dev` part at the end of the current version before the local part.
+
+The dev part is formatted that way `dev.{number}`. The `{number}` value is calculated from the current date with the following formula:
+
+$f(year, month, day) = year * 366 + month * 31 + day$
```

Comment:
Where do this formula come from?

It is not clear to me why is this an _improvement_ (as in the section title)? Is there a particular problem/issue with the current formula (besides the mistake on year factor)? 
If the answer is no, is probably better to open an issue to "fix" the dev format, proposing the alternatives that you listed here.

Reply:
> Where do this formula come from?

From my hat :tophat:

> It is not clear to me why is this an improvement ...

For me some formulas are simpler to use

> Is there a particular problem/issue with the current formula (besides the mistake on year factor)?

The problem is that value overlap between December and January:

```math
\begin{align}
f(y, m, d) = y * 366 + m * 31 + d \\
f(0, 12, 26) = f(1, 1, 1) = 398 \\
f(0, 12, 31) = f(1, 1, 6) = 403
\end{align}
```

Reply:
> > Where do this formula come from?
> 
> From my hat 🎩

So indeed, better try to find a more standard way. I am not sure that we really need the date information, so maybe just an incremental index.

> > Is there a particular problem/issue with the current formula (besides the mistake on year factor)?
> 
> The problem is that value overlap between December and January:

IMO this is acceptable for the moment. I suggest dealing with this subject in a separate issue (so we can discuss there what is the best naming scheme).

Reply:
Resolving the conversation as this is no longer relevant (see https://github.com/Scille/parsec-cloud/pull/6592)